File: PhotoFox.Model/PhotoAlbum.cs
﻿namespace PhotoFox.Model
{
    public class PhotoAlbum
    {
        public PhotoAlbum(
            string albumId, 
            string title, 
            string description, 
            string coverPhotoId,
            string folder,
            string sortOrder,
            bool isPublic)
        {
            this.AlbumId = albumId;
            this.Title = title;
            this.Description = description;
            this.CoverPhotoId = coverPhotoId;
            this.Folder = folder;
            this.SortOrder = sortOrder;
            this.IsPublic = isPublic;
        }

        public string AlbumId { get; }
        public string Title { get; }
        public string Description { get; }
        public string CoverPhotoId { get; }
        public string Folder { get; }
        public string SortOrder { get; }
        public bool IsPublic { get; }
    }
}

File: Photofox.Android/app/src/main/java/com/fionasapphire/photofox/model/PhotoAlbum.kt
package com.fionasapphire.photofox.model

/**
 * An album of photos in the main Photo Album list
 * @property albumId The Unique ID of the album
 * @property title The main display name
 * @property description The description of the album (currently unused)
 * @property image A reference to the image used as a cover photo
 */
data class PhotoAlbum(
    val albumId: String,
    val title: String,
    val description: String,
    val image: ImageReference
)